### DISCUSSION TITLE
When will support for 1.22.1 available?

### DISCUSSION THREAD

**Discussion by RatanVMistry** [eddycharly/terraform-provider-kops#424] [Q&A] [answered]
I am trying to import kops cluster which is provisioned by kosp 1.22.1. 
I can import most of the parts but it only gave error for "kops_cluster_updater" because of the kops version mismatch.

**Comment by eddycharly** [comment]
Hello, thanks for trying it out.
Unfortunately I have not added kOps 1.22 support yet.

I started https://github.com/eddycharly/terraform-provider-kops/pull/390 most notable change was related to the certs rotation iirc.  

I hope to be able to work on it soon 🤞

**Reply by RatanVMistry** [reply]
Any rough timeline for it?

**Reply by eddycharly** [reply]
I could try to cut an alpha release this week but I probably won’t have time to test it myself. Would you be interested in testing it if I do ?

**Reply by RatanVMistry** [reply]
yes sure. Do we have any procedures for testing?

**Reply by eddycharly** [reply]
Nothing well defined. At least it should be able to import an existing cluster, run a plan, an apply, etc…

Of course, should not be tested with a production cluster ;)

**Reply by RatanVMistry** [reply]
sure, I can try to import and others.

**Reply by eddycharly** [reply]
`v1.22.0-alpha.1` is being released, let me know if you have time to try it out 🙏 
For now i had to disable k8s certs management from terraform.

**Reply by eddycharly** [reply]
@RatanVMistry dunno if you could try it out but i released `v1.22.0-alpha.2` with a couple of fixes yesterday.

**Reply by argoyle** [reply]
Anything we can do to speed up support for 1.22 (and 1.23 as well)?

**Reply by eddycharly** [reply]
I think it’s all implemented except for the certificates part.
I’ll try to get a release next week, would you be able to test it ? I currently don’t have access to a test environment.

**Reply by argoyle** [reply]
That sounds perfect. I have my own cluster to try it out with and can spin up new clusters easily as well.

**Reply by eddycharly** [reply]
@argoyle did you try an alpha release before ? (`v1.22.0-alpha.3`)

**Reply by argoyle** [reply]
Not myself but I'm quite certain my colleague did.

**Reply by eddycharly** [reply]
Did it work ? I'm going to release `v1.22.0-alpha.4` based on latest kOps 1.22 version, if you can give it a try it would be great.

**Reply by argoyle** [reply]
I think so. We have a PR on one of hour TF modules at least. 😊 

I'll try alpha 4 during the day.

**Reply by eddycharly** [reply]
Great, I'm fixing ca certs support (it changed in kops 1.22) and cut the release.

**Comment by eddycharly** [comment, marked as answer]
@argoyle `v1.22.0-alpha.4` is available, let me know if you try it out.

**Reply by argoyle** [reply]
Trying now!

**Reply by argoyle** [reply]
Cluster rolling now (still using Kubernetes 1.21.11 this time around). First master just joined and seems to be working fine so far.

**Reply by eddycharly** [reply]
🤞

**Reply by eddycharly** [reply]
Verdict ?

**Reply by argoyle** [reply]
I had some trouble with the AWS LoadBalancer Controller but that is not a problem with the provider of course. Rolling again now upgrading to 1.22.8 but so far it feels solid. I would say you should be able to release it proper.

**Reply by eddycharly** [reply]
Thanks, I'll release it later today.
I started to add 1.23 support in #525. There are a couple of changes in the internal API though, not sure what is the best way to cope with this.
I will think about for one day or two and probably cut one pre release version later this week.

**Reply by argoyle** [reply]
Upgrade to K8S 1.22.8 went through without any problems at all.

**Reply by eddycharly** [reply]
`v1.22.0` has been published, please open issues if have troubles.
